pcap/pcapio/reader.go

```diff
@@ -34,6 +34,14 @@ type Reader interface {
 
 // NewReader returns a Reader by trying both pcap and pcap-ng formats.
 func NewReader(r io.Reader) (Reader, error) {
+	return NewReaderWithWarnings(r, nil)
+}
+
+// NewReaderWithWarnings returns a Reader by trying both pcap and pcap-ng formats
+// and arranges for warning messages to be sent over the given channel.  Different
+// pcap implementations can have out-of-spec peculiarities that can be tolerated
+// so we send warnings and try to okeep going.
+func NewReaderWithWarnings(r io.Reader, warningCh chan<- string) (Reader, error) {
```

Comment:
+1 for creating a new method rather than just tacking `warningCh` as an additional argument to `NewReader`